### PR TITLE
YAML: Use C bindings if available

### PR DESCRIPTION
**Pull request by cristinapauna**
C bindings offer a considerable speed increase, so use them when
available (i.e. if libyaml is installed).

While at it, explicitly use safe dump for reference-enabled YAML.

Is a partial fix for https://jira.opnfv.org/browse/ARMBAND-347

Signed-off-by: Cristina Pauna <cristina.pauna@enea.com>

### Previous Behavior
Remove this section if not relevant

### New Behavior
Remove this section if not relevant

### Tests written?

Yes/No

### Commits signed with GPG?

Yes/No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: salt/config/__init__.py [hunk 1]
```diff
@@ -46,6 +46,7 @@
 except ImportError:
     HAS_PSUTIL = False
 
+_SafeDumper = yaml.CSafeDumper if yaml.__with_libyaml__ else yaml.SafeDumper
```
Comment (isbm) [inline salt/config/__init__.py]: Unused.

File: salt/config/__init__.py [hunk 2]
```diff
@@ -2138,7 +2139,7 @@ def _read_conf_file(path):
     log.debug('Reading configuration from %s', path)
     with salt.utils.files.fopen(path, 'r') as conf_file:
         try:
-            conf_opts = salt.utils.yaml.safe_load(conf_file) or {}
+            conf_opts = salt.utils.yaml.load(conf_file, Loader=_SafeLoader) or {}
```
Comment (isbm) [inline salt/config/__init__.py]: `_SafeLoader` is not defined.

**Review by isbm — CHANGES_REQUESTED** [review]
I am not sure this is actually working. Please write a Unit Test for this.

**Comment by cristinapauna** [comment]
Hi,

I was testing on a different branch and didn’t pay attention to the differences. I am aware of the issue and will update the patch shortly.

Thanks,
Cristina

**Comment by cristinapauna** [comment]
I need to reconsider the fix using the salt yaml library.

**Pull request closed without merging**